pypowsybl/dynamic.py

```diff
@@ -35,46 +36,57 @@ def __init__(self) -> None:
-                                                          "parameter_set_id": [dynamic_param],
-                                                          "mapping_type": [MappingType.GENERATOR_SYNCHRONOUS_THREE_WINDINGS]}))
+        self.add_all_dynamic_mappings(static_id=static_id,
+                                      parameter_set_id=dynamic_param,
+                                      mapping_type=MappingType.GENERATOR_SYNCHRONOUS_THREE_WINDINGS)
 
     def add_generator_synchronous_three_windings_proportional_regulations(self, static_id: str, dynamic_param: str) -> None:
-        self.add_all_dynamic_mappings(_pd.DataFrame(data={"static_id": [static_id],
-                                                          "parameter_set_id": [dynamic_param],
-                                                          "mapping_type": [MappingType.GENERATOR_SYNCHRONOUS_THREE_WINDINGS_PROPORTIONAL_REGULATIONS]}))
+        self.add_all_dynamic_mappings(static_id=static_id,
+                                      parameter_set_id=dynamic_param,
+                                      mapping_type=MappingType.GENERATOR_SYNCHRONOUS_THREE_WINDINGS_PROPORTIONAL_REGULATIONS)
 
     def add_generator_synchronous_four_windings(self, static_id: str, dynamic_param: str) -> None:
-        self.add_all_dynamic_mappings(_pd.DataFrame(data={"static_id": [static_id],
-                                                          "parameter_set_id": [dynamic_param],
-                                                          "mapping_type": [MappingType.GENERATOR_SYNCHRONOUS_FOUR_WINDINGS]}))
+        self.add_all_dynamic_mappings(static_id=static_id,
+                                      parameter_set_id=dynamic_param,
+                                      mapping_type=MappingType.GENERATOR_SYNCHRONOUS_FOUR_WINDINGS)
 
     def add_generator_synchronous_four_windings_proportional_regulations(self, static_id: str, dynamic_param: str) -> None:
-        self.add_all_dynamic_mappings(_pd.DataFrame(data={"static_id": [static_id],
-                                                          "parameter_set_id": [dynamic_param],
-                                                          "mapping_type": [MappingType.GENERATOR_SYNCHRONOUS_FOUR_WINDINGS_PROPORTIONAL_REGULATIONS]}))
+        self.add_all_dynamic_mappings(static_id=static_id,
+                                      parameter_set_id=dynamic_param,
+                                      mapping_type=MappingType.GENERATOR_SYNCHRONOUS_FOUR_WINDINGS_PROPORTIONAL_REGULATIONS)
 
     def add_current_limit_automaton(self, static_id: str, dynamic_param: str, branch_side: BranchSide) -> None:
         _pp.add_current_limit_automaton(
             self._handle, static_id, dynamic_param, branch_side.value)
 
-    def add_all_dynamic_mappings(self, df: _pd.DataFrame) -> None:
+    def add_all_dynamic_mappings(self, df: _pd.DataFrame = None, **kwargs: _ArrayLike) -> None:
```

Comment:
I have issues to type kwargs here.
Using the same typing as in network.py does not work since I allow enum values like : 
```
mapping.add_all_dynamic_mappings(static_id="LOAD1", parameter_set_id="LAB", mapping_type=MappingType.LOAD_ALPHA_BETA)
```